File: BRLADDER.cpp
#include <bits/stdc++.h>
using namespace std;

int main(int argc, char const *argv[])
{
    int t, a, b;
    cin>>t;
    while(t--)
    {
        cin>>a>>b;
        int temp = min(a,b);
        a = max(a, b);
        b = temp;
        if(b%2==1)
        {
            // odd
            if(a-b < 3)
                cout<<"YES\n";
            else
                cout<<"NO\n";
        }
        else {
            // even
            if(a-b == 2)
                cout<<"YES\n";
            else
                cout<<"NO\n";
        }
    }
    return 0;
}

File: BRLADDER.java
import java.lang.*;
import java.io.*;
import java.util.*;

class BRLADDER
{
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int t = scanner.nextInt();
        while(t>0)
        {
            t--;
            int a = scanner.nextInt();
            int b = scanner.nextInt();
            int temp = Math.max(a, b);
            b = Math.min(a, b);
            String ans = "";
            a = temp-b;
            if(b%2==0)
            {
                if(a==2)
                    ans = "YES";
                else
                    ans = "NO";
            }
            else 
            {
                if(a<3)
                    ans ="YES";
                else
                    ans = "NO";
            }
            System.out.println(ans);
        }
        scanner.close();
    }
}